### ISSUE TITLE
Feature Request: Support warm-start/fine-tuning from pretrained EAGLE3 models

### ISSUE THREAD

**Issue by VincentG1234** [vllm-project/speculators#269]
### Feature Request: Support warm-start/fine-tuning from pretrained EAGLE3 models

**Is your feature request related to a problem? Please describe.**
Currently, `scripts/train.py` only supports training EAGLE3 models from scratch. To fine-tune an already trained EAGLE3 model (e.g., published on HuggingFace Hub), users must manually extract vocab mappings and weights, which is cumbersome and error-prone.

**Describe the solution you'd like**
Add support for warm-start/fine-tuning from pretrained EAGLE3 models with a new `--pretrained-model-path` argument that:- Accepts both local paths and HuggingFace Hub model IDs- Automatically loads vocab mappings (`d2t`/`t2d`) from the pretrained model- Automatically derives `draft_vocab_size` from loaded mappings- Loads model weights while keeping optimizer/scheduler states fresh

**Expected behavior**
Users should be able to fine-tune a pretrained EAGLE3 model with a simple command:

python scripts/train.py \  
--verifier-name-or-path meta-llama/Llama-3.1-8B-Instruct \  
--pretrained-model-path RedHatAI/Llama-3.1-8B-Instruct-speculator.eagle3 \ 
 --data-path ./new_domain_data \  
--epochs 5 \  
--lr 5e-5

**Use cases**
Domain adaptation: Fine-tune a general EAGLE3 model on domain-specific data
Continued training: Resume training with new data without full checkpoint state
Model iteration: Start from a community model and adapt to specific requirements
Research: Experiment with different fine-tuning strategies on published models

**Status**
🚧 PR in progress - #268
